uk/arch: Add `ukarch_fetch_sub`

razvanvirtan opened the pull request:
### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [x] Updated relevant documentation.


### Base target

 - Architecture(s): [`x86_64`, arm64]
 - Platform(s): [N/A]
 - Application(s): [N/A]


### Additional configuration

### Description of changes

Since we are using wrappers over the GNU atomic builtins (see [this header](https://github.com/unikraft/unikraft/blob/staging/include/uk/arch/atomic.h), we need to complete them with a wraper for `__atomic_fetch_sub `. For instance, this is required in [this PR](https://github.com/unikraft/unikraft/pull/801).

razvanvirtan committed:
uk/arch: Add `ukarch_fetch_sub`

Add wrapper for the `__atomic_fetch_sub` GNU builtin.

Signed-off-by: Razvan Virtan <razvanvirtan@gmail.com>

marcrittinghaus commented on include/uk/arch/atomic.h:
```diff
@@ -56,6 +56,12 @@ extern "C" {
 #define ukarch_fetch_add(src, value) \
 	__atomic_fetch_add(src, value, __ATOMIC_SEQ_CST)
 
+/**
+ * Perform a atomic fetch and sub operation.
+ */
+#define ukarch_fetch_sub(src, value) \
+	__atomic_fetch_sub(src, value, __ATOMIC_SEQ_CST)
+
```
Hey @razvanvirtan,

thanks for adding this function. Looking at 5b3d85f29292c45dde0f18b656ef9224c1655a0c and the fact that `inc`/`dec` are grouped, I would also group this one accordingly.

Please also fix the `a atomic` typo (-> `an atomic`)

marcrittinghaus replied:
I just recognize that the definition of `ukarch_inc` uses `ukarch_fetch_add` for the implementation. For consistency, I suggest to also change `ukarch_dec` to use your new definition as a separate commit.

razvanvirtan committed:
uk/arch: Make `ukarch_dec` use `ukarch_fetch_sub`

Change the `ukarch_dec` definition so that it uses `ukarch_fetch_sub`,
being consistent with the `ukarch_inc` implementation.

Signed-off-by: Razvan Virtan <virtanrazvan@gmail.com>

razvanvirtan commented:
@marcrittinghaus Thanks for your suggestions! I have updated the PR.

eduardvintila reviewed (approved):
Hey, @razvanvirtan! Thanks for adding this; looks good to me!

Reviewed-by: Eduard Vintilă <eduard.vintila47@gmail.com>

michpappas commented:
@razvanvirtan looks good to me. In the description please rephrase without the commit hash, as it will change as soon github appends review / approve tags to the commit message.
```
uk/arch: Make ukarch_dec use ukarch_fetch_sub

In dbdca13, we introduced the `ukarch_fetch_sub` macro.
...
```
Other than that I'm ready to merge as soon as @marcrittinghaus ACKs that he's happy that his review comments have been addressed.

marcrittinghaus reviewed (approved):
Regarding my changes, I am fine 😃 @razvanvirtan, thanks for addressing!

I will already add my tag so we can merge a bit faster when the commit message has been updated.

Reviewed-by: Marc Rittinghaus <marc.rittinghaus@unikraft.io>

razvanvirtan commented:
@michpappas Done

michpappas commented:
@razvanvirtan thanks

Approved-by: Michalis Pappas <michalis@unikraft.io>
